java/ql/src/experimental/Security/CWE/CWE-094/GroovyInjectionBad.java

```diff
@@ -0,0 +1,49 @@
+public class GroovyInjection extends HttpServlet {
+
+    protected void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        try {
+            String script = request.getParameter("script");
+            final GroovyClassLoader classLoader = new GroovyClassLoader();
+            Class groovy = classLoader.parseClass(script);
+            GroovyObject groovyObj = (GroovyObject) groovy.newInstance();
+
+        } catch (Exception e) {
```

Comment:
You could make these examples briefer by removing the try/catch block and showing them in the context of a method that throws `Exception`.